archiver/links.py

```diff
@@ -74,6 +74,14 @@ def validate_links(links):
 
     return list(links)
 
+def new_links(all_links, existing_links):
+    """
+    Return all links which are in the all_links but not in the existing_links.
+    This is used to determine which links are new and not indexed jet. Set the
+    ONLY_NEW environment variable to activate this filter mechanism.
+    """
+    existing_urls = list(map(lambda l: l['url'], existing_links))
+    return list(filter(lambda l: l['url'] not in existing_urls, all_links))
```

Comment:
I think this is still O(n^2) because it'll iterate though all of `existing_urls` to find a match, instead of doing an O(1) lookup into a `dict`/`set` of urls.

Here's an O(n) solution that could work:

```suggestion
existing_urls = {l['url'] for l in existing_links}
return [link for link in all_links if link['url'] not in existing_urls]
```

Reply:
Ok, I am not a Python programmer by any means. But I am sure my solution is O(2n), which is essentially O(n). I've actually benchmark this to see if the execution time is linear. Surprisingly, you are right, my solution IS in fact exponential, but if I remove the list() wrapper from the first line, it IS linear. I do not know the internals of map and filter in Python, but without the list() wrapper, 200.000 Bookmarks will take ~6 seconds on my machine (same as your solution).

So, map and filter runs in O(n) (Still not sure what magic the list() thing is doing internally). But nevertheless, I would like to use your solution, because it matches the coding style of the rest of the code.